### PR TITLE
Fix issue where using DisableNoise or disabling add_noise results in non-deterministic generations

### LINKED ISSUE #2833: No seed gets set in custom/advanced samplers when add_noise is disabled
the common sampler code has:

```python
    if disable_noise:
        noise = torch.zeros(latent_image.size(), dtype=latent_image.dtype, layout=latent_image.layout, device="cpu")
    else:
        batch_inds = latent["batch_index"] if "batch_index" in latent else None
        noise = comfy.sample.prepare_noise(latent_image, seed, batch_inds)
```

`prepare_noise` is what sets the global RNG that various thing such as ancestral samplers use, so when `add_noise` is disabled this means no seed gets set before sampling and the RNG is an undefined state. in other words, if you turn off `add_noise` and use a sampler like `euler_a` it's impossible to reproduce generations.

it's easy to reproduce: just use an advanced or custom sampler, set a fixed seed, turn off `add_noise` and use `euler_a`. start generating and look at the preview. abort it at 50% or so and restart, you will get a completely new generation each time.

is there any reason not to just do `torch.manual_seed(seed)` in the `disable_noise` code path to ensure the RNG is in a defined state based on the seed the user set?

i made a simple work around ( https://github.com/blepping/ComfyUI-bleh?#blehforceseedsampler ) but it would be nice if that wasn't necessary.

### PR DESCRIPTION
**Pull request by blepping**
ComfyUI currently has a bug where disabling `add_noise` in samplers or using a `DisableNoise` node simply does not seed the RNG at all, resulting in non-deterministic generations. #4518 aims to fix that problem by actually seeding the RNG even when adding noise is disabled. this pull takes a slightly different approach: we throw an error when the user attempts to disable noise rather than proceeding anyway and erroneously ignoring the seed.

this pull could also be changed to simply remove the `DisableNoise` node and `add_noise` parameters but i thought it would be a bit more friendly to let people load existing workflows.

closes #2833

### CODE REVIEW

**Comment by catboxanon** [comment]
This would break compatibility with https://github.com/BlenderNeko/ComfyUI_Noise. I'm sure your intent with this PR is just to get something merged at all to avoid this bug though, which should've have had the fix merged months ago by now.

**Comment by blepping** [comment]
> This would break compatibility with https://github.com/BlenderNeko/ComfyUI_Noise. 

and my own nodes also.

> I'm sure your intent with this PR is just to get something merged at all to avoid this bug though

eh, i knew from the beginning there was no chance of it getting merged. it's not really serious, but i genuinely think merging it would be better than the status quo where disabling `add_noise` in the built-in samplers is just silently broken.

at least just preventing the ability to disable `add_noise` would push people toward using sampler nodes that actually work reliably.
